nova/db/sqlalchemy/models.py

```diff
@@ -1604,7 +1604,7 @@ class ConsoleAuthToken(BASE, NovaBase):
     console_type = Column(String(255), nullable=False)
     host = Column(String(255), nullable=False)
     port = Column(Integer, nullable=False)
-    internal_access_path = Column(String(255))
+    internal_access_path = Column(Text)
```

Comment:
IIRC, the difference between text and varchar in postgres was, that text is saved as blob in an extra column, while varchar is saved right in the table.

Do we want that extra lookup here? Does it matter? Could we upstream this or is `String(400)` easier?

Reply:
Is it customer-controlled data? If yes, then it should definitely be Text (or at least, way more than 255 chars).

Reply:
Those tokens are temporary and I don't think we'll get over a 100 active console sessions ever in the largest regions.

Do you mean VARCHAR(400)? 
- that is an option, but what if that is not enough?
- also won't work on older versions of mysql.

Reply:
I mean `String(400)` as this should be translated by SQLAlchemy to `VARCHAR(400)`.
we have a maximum length on the VM name afaik. nova always shortens the names, because of that. so the only thing that could extend it more, would be the path of the datastore. having additional 145 characters I would think is enough, but we could compute how much we actually have after removing the max name * 2 + len(.vmx) + 2 (for 2* /)

Reply:
https://github.com/sapcc/nova/blob/42f8679570c214bf775c34259dbd8737b844b37a/nova/virt/vmwareapi/vm_util.py#L1668

Reply:
```
2 * (41 + 37 + 1) + 4 + 2
164
```
without the display_name the max here is 80
+ the other part of the JSON
```
len('{"ticket": "528ef836-34f0-343f-413f-5c1c7792f960", "thumbprint": "c42e74a70488a176818689e87f5618e6669f36c8", "cfgFile": "')
121
```
+2 for `"}`

```
2 * (36 + 1) + 4 + 2
80
121+2+80
203
121+2+164
287
400 - 287
113
```
so we have 113 chars left for the path to the datastore. I think that should be enough.

Reply:
There is not a single column in nova which would use String(400), however there are quite a few that use Text.
Up to you @joker-at-work.

Reply:
You're probably right, that it doesn't matter.